File: internal/model/video.go
package model

import "time"

type Video struct {
	ID          string        `json:"id"`
	Title       string        `json:"title"`
	URL         string        `json:"url"`
	Date        string        `json:"date"`
	JPG         string        `json:"jpg"`
	PicturesDir string        `json:"pictures_dir"`
	Actors      []Actor       `json:"actors"`
	Tags        []Tag         `json:"tags"`
	Formats     []VideoFormat `json:"formats"`
	IsFavorite  bool          `json:"is_favorite"`
	CreatedAt   time.Time     `json:"created_at"`
	UpdatedAt   time.Time     `json:"updated_at"`
}

type Actor struct {
	ID   int    `json:"id"`
	Name string `json:"name"`
}

type Tag struct {
	ID   int    `json:"id"`
	Name string `json:"name"`
}

type VideoFormat struct {
	ID       int    `json:"id"`
	Name     string `json:"name"`
	FilePath string `json:"file_path"`
}

type VideoQueryParams struct {
	Page     int
	PerPage  int
	Query    string
	Tag      string
	Actor    string
	DateFrom string
	DateTo   string
	Sort     string
	Favorite bool
	HasVideo bool
}

type VideoListResult struct {
	Data       []Video `json:"data"`
	Total      int     `json:"total"`
	Page       int     `json:"page"`
	PerPage    int     `json:"per_page"`
	TotalPages int     `json:"total_pages"`
}

File: frontend/src/types/video.ts
export interface Video {
  id: string;
  title: string;
  url: string;
  date: string;
  jpg: string;
  pictures_dir: string;
  actors: Actor[];
  tags: Tag[];
  formats: VideoFormat[];
  rating: number;
  created_at: string;
  updated_at: string;
}

export interface Actor {
  id: number;
  name: string;
}

export interface Tag {
  id: number;
  name: string;
}

export interface VideoFormat {
  id: number;
  name: string;
  file_path: string;
}

export interface PaginatedResponse<T> {
  data: T[];
  total: number;
  page: number;
  per_page: number;
  total_pages: number;
}
